### PR TITLE
#161967096 Article pagination 

### PR DESCRIPTION
**Pull request by bettblake08**
**What does this PR do?**
-  Paginate article on the homepage

**Description of tasks to be completed?**
- Create a landing section of the homepage
- Create a top stories section on the homepage
- Create a recent stories section on the homepage

**How should this be manually tested?**
- Clone the repo and cd into the project root folder
- Switch to the relevant branch `git checkout ft-article-pagination-161967096`
- Run `npm install` to update your packages.
- Start the server with `npm start`
- In your favorite browser, navigate to `http://localhost:3000`

**What are the relevant pivotal tracker stories?**
- [#161967096](https://www.pivotaltracker.com/story/show/161967096)

**Screenshots**
image
image

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: __mocks__/redux-persist/integration/react.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// jest.mock("redux-persist/integration/react", () => ({
```
Comment (justmesam) [inline __mocks__/redux-persist/integration/react.js]: why is the file commented? also its bad practice to push commented code. 
What does the file do?

File: src/api/axiosConfig.js [hunk 2]
```diff
@@ -1,8 +1,13 @@
 import axios from "axios";
 
-const instance = axios.create({
+const apiDev = axios.create({
```
Comment (justmesam) [inline src/api/axiosConfig.js]: Why was this renamed? also have you consulted the others on this since everyone is still using instance on their work?

File: src/components/containers/headers/__test__/headers.test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { shallow } from "enzyme";
+import HeaderComponent from "../index";
+
+describe("General HeaderComponent :", () => {
```
Comment (justmesam) [inline src/components/containers/headers/__test__/headers.test.js]: this file doesn't test anything, we would rather have it removed.

**Review by justmesam — CHANGES_REQUESTED** [review]

File: __mocks__/redux-persist/integration/react.js (hunk 1 above)
Comment (bettblake08) [reply __mocks__/redux-persist/integration/react.js]: I kept the file so I may debug the setup of the test environment later on. I'll remove it since its bad practice.

File: src/api/axiosConfig.js (hunk 2 above)
Comment (bettblake08) [reply src/api/axiosConfig.js]: We are exporting the instance as default. My assumption is that when we export it as a default, the name does not matter when importing.

File: src/components/containers/headers/__test__/headers.test.js (hunk 3 above)
Comment (bettblake08) [reply src/components/containers/headers/__test__/headers.test.js]: I have decided to add more tests. Please wait for the update.

File: src/api/axiosConfig.js (hunk 2 above)
Comment (justmesam) [reply src/api/axiosConfig.js]: 👍

**Review by justmesam — APPROVED** [review]

**Review by denomash — APPROVED** [review]
👍

**Review by Darius-Ndubi — APPROVED** [review]
👍

**Review by davishooly — APPROVED** [review]

**Review by jillo-abdullahi — DISMISSED** [review]
Hey @bettblake08 I get an error on `src/api/articleAPI.js:78` when on localhost. It can't resolve the response status of 200. What could be the problem?

File: src/components/views/articles/article1.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
```
Comment (justmesam) [inline src/components/views/articles/article1.js:1]: What is the difference between this file and `article2.js` ?

File: src/components/containers/landingSection.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+  componentDidMount() {
+    const { actions } = this.props;
+    actions.article.getArticles("landingSection", true);
+  }
+
+  render() {
+    const { articles } = this.props;
+    const { slider } = this.state;
+    const articlesList = [[], [], []];
```
Comment (justmesam) [inline src/components/containers/landingSection.js:43]: Why is this array declared this way?

**Review by justmesam — CHANGES_REQUESTED** [review]

File: src/components/views/articles/article1.js (hunk 4 above)
Comment (bettblake08) [reply src/components/views/articles/article1.js:1]: They are two different ways of displaying articles as items.

File: src/components/containers/landingSection.js (hunk 5 above)
Comment (bettblake08) [reply src/components/containers/landingSection.js:43]: It's a list of lists. The idea is to output 3 articles in 3 separate slides. The slider is currently not working so I had left that to work on in the coming new PRs.

File: src/components/views/articles/article1.js (hunk 4 above)
Comment (justmesam) [reply src/components/views/articles/article1.js:1]: Give more context.

Comment (bettblake08) [reply src/components/views/articles/article1.js:1]: There are two different article components on the homepage. One type is in the landing section (article1.js) and the others are in the recent stories section and the top stories section (article2.js). The same article2.js is being used in the discover.

**Review by Bryan-Cee — APPROVED** [review]

**Review by justmesam — APPROVED** [review]
